pkg/suggestion/v1beta1/skopt/service.py

```diff
@@ -86,3 +95,38 @@ def convertAlgorithmSpec(algorithm_spec):
             elif s.name == "random_state":
                 optimizer.random_state = int(s.value)
         return algorithm_spec.algorithm_name, optimizer
+
+    @classmethod
+    def validate_algorithm_spec(cls, algorithm_spec):
+        algo_name = algorithm_spec.algorithm_name
+
+        if algo_name == "bayesianoptimization":
+            return cls._validate_bayesianoptimization_setting(algorithm_spec.algorithm_settings)
+        else:
+            return False, "unknown algorithm name {}".format(algo_name)
+
+    @classmethod
+    def _validate_bayesianoptimization_setting(cls, algorithm_settings):
+        for s in algorithm_settings:
+            try:
+                if s.name == "base_estimator":
+                    if s.value not in ["GP", "RF", "ET", "GBRT"]:
+                        return False, f"base_estimator {s.value} is not supported in katib"
```

Comment:
Should we point to specific algorithm here ?
```suggestion
                        return False, f"base_estimator {s.value} is not supported in Bayesian optimization"
```

Reply:
It would be better! Thanks